instance/models/rabbitmq.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2016 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Definition of RabbitQM-specific models.
+"""
+from django.db import models
+from django_extensions.db.models import TimeStampedModel
+
+from instance.models.utils import ValidateModelMixin
+
+
+class RabbitMQUser(ValidateModelMixin, TimeStampedModel):
+    """
+    A model representing a single RabbitMQ user
+    """
+    username = models.CharField(max_length=32, unique=True)
```

Comment:
We're enforcing username uniqueness, which means an IntegrityError will be raised if we get a duplicate. It's not likely to happen, but are we handling such an error and generating a different name if necessary? Alternatively, can we check to see if a user with a particular username exists before going to create a new one? (abstract the creation into a method that performs the check)